### ISSUE TITLE
Read inside table cells

### ISSUE THREAD

**Issue by kiarash3tar** [PHPOffice/PHPWord#768]
Hi;
First of all I have to say thank you for this repo.

I want to read a `docx` file that contains few tables and then insert data from every table to a **MYSQL TABLE**.

This is my code:

```
require_once 'vendor/autoload.php';
$source = 'testDoc.docx';
$phpWord = \PhpOffice\PhpWord\IOFactory::load($source);

$sections = $phpWord->getSections();

```
I don't know how to get tables texts `cell by cell` and manipulate them.

Would you please describe more about read `docx` file?

_**Thans in Advance**_

**Comment by pv451** [comment]
well, how about this code?

	
	foreach ($sections[0]->getElements() as $el)
	{
		
		if ($el instanceof PhpOffice\PhpWord\Element\Table)
		{
		echo 'new table <br><br>';
		foreach ($el->getRows() as $row)
		{
			foreach ($row->getCells() as $cell)
			{
				echo $cell->getElements()[0]->getText().'***';
			}
			echo '<br>';
		}}
	}

**Comment by kiarash3tar** [comment]
Thanks
This works, but I have this error for one of tables and the code will break from this poitn:

```Uncaught Error: Call to undefined method PhpOffice\PhpWord\Element\TextBreak::getText() in C:\xampp\htdocs\test2\index.php:41```

**Comment by pv451** [comment]
Yep, this could be. Also it will shoot if you have some complex data in cell.
Surround getText() with instanceof check. like that
```php
if ($cell->getElements()[0] instanceof PhpOffice\PhpWord\Element\Text)
{
    echo $cell->getElements()[0]->getText().'***';
}
```

**Comment by R1sen** [comment]
Hi, thanks for this repository and sorry for my bad English. You really help me! But from someone colum cant get data, how i can get all data? This column has  normal text  no more 5000 symbol. Coding is utf 8. Instead, I get a text *** that means empty. Thanks!

**Comment by pv451** [comment]
Well, it seems than your cell have a complex structure inside. You should check what elements your cell contains. Try something like replacing
```php
 echo $cell->getElements()[0]->getText().'***';
```
to 

```php
 print_r($cell->getElements());
```

**Comment by R1sen** [comment]
Спасибо за столь быстрый ответ, я попробовать его сейчас!

**Comment by pv451** [comment]
@R1sen Ага, отпишись по результатам.

**Comment by R1sen** [comment]
Hm. My code:
<?php
    require_once 'PHPWord-develop/vendor/autoload.php';
    $source = 'WR290316.docx';
    $phpWord = \PhpOffice\PhpWord\IOFactory::load($source);

    $sections = $phpWord->getSections();
    ob_start();
    foreach ($sections[0]->getElements() as $el)
    {

        if ($el instanceof PhpOffice\PhpWord\Element\Table)
        {
            echo '<table width="1">';

            foreach ($el->getRows() as $row)
            {
                echo "<tr>";
                foreach ($row->getCells() as $cell)
                {
                    if ( $cell -> getElements()[ 0 ] instanceof  PhpOffice\PhpWord\Element\ Text )
                    {
                        echo  "<td>".$cell -> getElements()[ 0 ] -> getText() ."</td>" ;
                    }else
                    {
                        $content = $cell->getElements();
                        print_r($content);

                    }
                }
                echo "</tr>";
            }
            echo '</table>';
        }
    }
    $data = ob_get_clean();
    
    
    ?>
    
    <!doctype html>
    <html>
    <head>
        <meta charset="UTF-8">
        <title>title</title>
    
    </head>
    <body>
    
    <table>
        <?=$data?>
    </table>
    </body>
    </html>
if I do, that's the case print_r($cell->getElements()); , then nothing is displayed white screen with no errors. strangely

**Comment by pv451** [comment]
What is that `<?=$data?>`? can you attach WR290316.docx?

**Comment by R1sen** [comment]
I save table(in html)  in var $data

**Comment by R1sen** [comment]
thx i did it.

2016-03-29 14:02 GMT+03:00 pv451 <notifications@github.com>:

**Comment by pv451** [comment]
Well, as i said there is more complex content of the Cell, there TextRun instead of Text. You may solve this problem with additional processing like that
```php
$source = 'WR290316.docx';
$phpWord = \PhpOffice\PhpWord\IOFactory::load($source);

$sections = $phpWord->getSections();	
	
	foreach ($sections[0]->getElements() as $el)
{

    if ($el instanceof PhpOffice\PhpWord\Element\Table)
    {
    echo '<table border =1 >';
    foreach ($el->getRows() as $row)
    {
		echo '<tr>';
        foreach ($row->getCells() as $cell)
        {
			echo '<td>';
			foreach ($cell->getElements() as $cEl)
				if ($cEl instanceof PhpOffice\PhpWord\Element\Text)
				{
					echo $cEl->getText() .'<br>';
				}
				elseif ($cEl instanceof PhpOffice\PhpWord\Element\TextRun){
					if (count($cEl->getElements())>0 and $cEl->getElements()[0] instanceof PhpOffice\PhpWord\Element\Text)
					{
						echo $cEl->getElements()[0]->getText();
					}
				}
			echo '</td>';
        }
        echo '</tr>';
    }
	echo '</table>';
	}
}
```

**Comment by R1sen** [comment]
Thanks for help with my problem, i try it and comment result.

**Comment by kaoukoui** [comment]
Undefined offset: 0 : foreach ($sections[0]->getElements() as $el)
Fatal error: Call to a member function getElements() on a non-object 
plz help me
